ws-client/src/client.rs

```diff
@@ -134,15 +157,16 @@ impl WsClient {
 	/// Initializes a new WebSocket client
 	///
 	/// Fails when the URL is invalid.
-	pub async fn new(remote_addr: impl AsRef<str>, config: WsConfig) -> Result<Self, Error> {
-		let (sender, receiver) = jsonrpc_transport::websocket_connection(remote_addr.as_ref())
+	pub async fn new(config: WsConfig) -> Result<WsClient, Error> {
+		let (sender, receiver) = jsonrpc_transport::websocket_connection(config.clone())
 			.await
 			.map_err(|e| Error::TransportError(Box::new(e)))?;
 
-		let (to_back, from_front) = mpsc::channel(config.request_channel_capacity);
+		let channel_capacities = config.channel_capacities;
+		let (to_back, from_front) = mpsc::channel(channel_capacities.request);
```

Comment:
What happens when the capacity is set to `0`? Should we force the config to be `NonZeroUsize`, or default to either unbound channel (I guess that will require some extra generic mess here) or `usize::MAX`.

Reply:
That's a great question, according to the docs

> The channel's capacity is equal to buffer + num-senders.

So, in that case, it will be `1` which I think is fine. I was thinking about the `NonZero` before but it makes the API quite annoying to use. Thoughts?

Reply:
Ah, should be fine then. I'd just note in doc comment that the actual limit is 1 + this.

Reply:
I updated the documentation of `WsConfig` struct, ok?